### ISSUE TITLE
image2 404 error

### ISSUE THREAD

**Issue by abhinavnair** [django-ckeditor/django-ckeditor#310]
In my django app, image2 doesn't load. I have added it in the extraPlugin.

But the page shows this error - /static/ckeditor/ckeditor/plugins/image2/plugin.js?t=G6DE HTTP/1.1" 404

If I remove image2 from the extrasPlugin, the ckeditor does load with image plugin.

I checked my assets folder and I can't find image2 folder under ckeditor plugins folder

**Comment by riklaunim** [comment]
is it an additional external plugin to ckeditor?

**Comment by abhinavnair** [comment]
CKEDITOR_CONFIGS = {
    'app_ckeditor': {
        'toolbar': 'Full',
        'height': 300,
        'width': '100%',
        'uiColor': '#FAFAFA',
        'resize_enabled': False,
        'toolbar_Full': [
            ['Source', '-', 'NewPage', 'Preview', '-', 'Templates', 'Print'],
            ['Cut', 'Copy', 'Paste', 'PasteText', 'PasteFromWord', '-', 'Undo', 'Redo'],
            ['Link', 'Remove Link', 'Reference'],
            ['Image', 'Flash', 'Table', 'HorizontalRule'],
            '/',
            ['Bold', 'Italic', 'Underline', 'Strike', 'Remove Format', 'SpellChecker', 'Styles', 'Format'],
            ['TextColor', 'BGColor'],
            ['Smiley', 'SpecialChar'],
        ],
        'extraPlugins': 'image2',
    },
}

This is my configuration

**Comment by abhinavnair** [comment]
I downloaded image2 seperately and put it in my ckeditor plugins folder inside my asset, and it worked. But now it says that **widgets** not found.

**Comment by riklaunim** [comment]
That's the ~default config for extraPlugins:
```
'extraPlugins': ','.join(
            [
                # your extra plugins here
                'div',
                'autolink',
                'autoembed',
                'embedsemantic',
                'autogrow',
                # 'devtools',
                'widget',
                'lineutils',
                'clipboard',
                'dialog',
                'dialogui',
                'elementspath'
            ]),
```

**Comment by abhinavnair** [comment]
Still not working. I am using django-ckeditor 5.1.0, django 1.9.6, python 2.7.11. Is had anything to do with the versions error?

**Comment by riklaunim** [comment]
it's something with that ckeditor config.

**Comment by abhinavnair** [comment]
I changed the config settings to just this much:-

```
CKEDITOR_CONFIGS = {
    'pem_ckeditor': {
        'extraPlugins': ','.join(
            [
                'image2',
            ]),
    }
}
```

Still collectstatic not generating the image2 or widget files

**Comment by abhinavnair** [comment]
Is it possible to define where collectstatic copies the files to?
because I can do a bower_install of ckeditor and copy the files using collectstatic.

But the issue is that the ckeditor is installed in ckeditor/
But django-ckeditor installs files and search files from ckeditor/ckeditor/

So is it possible to check from ckeditor/ or install to ckeditor/ instead of ckeditor/ckeditor/ ?

**Comment by riklaunim** [comment]
if you mimic django-ckeditor directory structure then yes, you can put files in the same location, or add plugins.

**Comment by zkf** [comment]
The directory `django-ckeditor/ckeditor/static/ckeditor/ckeditor/plugins/image2`
 went missing after commit `25e60c12 Update to CkEditor 4.5.10 full package. `

**Comment by riklaunim** [comment]
Should be fixed in 5.1.1
